deps/rabbit/src/rabbit_global_counters.erl

```diff
@@ -0,0 +1,183 @@
+                },
+                {
+                    messages_delivered_consume_manual_ack_total, ?MESSAGES_DELIVERED_CONSUME_MANUAL_ACK, counter,
+                    "Total number of messages delivered to consumers using basic.consume with manual acknowledgment"
+                },
+                {
+                    messages_delivered_consume_auto_ack_total, ?MESSAGES_DELIVERED_CONSUME_AUTO_ACK, counter,
+                    "Total number of messages delivered to consumers using basic.consume with automatic acknowledgment"
+                },
+                {
+                    messages_delivered_get_manual_ack_total, ?MESSAGES_DELIVERED_GET_MANUAL_ACK, counter,
+                    "Total number of messages delivered to consumers using basic.get with manual acknowledgment"
+                },
+                {
+                    messages_delivered_get_auto_ack_total, ?MESSAGES_DELIVERED_GET_AUTO_ACK, counter,
+                    "Total number of messages delivered to consumers using basic.get with automatic acknowledgment"
+                },
+                {
+                    messages_get_empty_total, ?MESSAGES_GET_EMPTY, counter,
+                    "Total number of times basic.get operations fetched no message"
+                },
+                {
+                    messages_redelivered_total, ?MESSAGES_REDELIVERED, counter,
+                    "Total number of messages redelivered to consumers"
+                },
+                {
+                    messages_acknowledged_total, ?MESSAGES_ACKNOWLEDGED, counter,
+                    "Total number of messages acknowledged by consumers"
+                }
+            ]).
+
+boot_step() ->
+    init([{protocol, amqp091}]),
+    init([{protocol, amqp091}, {queue_type, rabbit_classic_queue}]),
+    init([{protocol, amqp091}, {queue_type, rabbit_quorum_queue}]),
+    init([{protocol, amqp091}, {queue_type, rabbit_stream_queue}]).
+
+init(Labels = [{protocol, Protocol}, {queue_type, QueueType}]) ->
+    _ = seshat_counters:new_group(?MODULE),
+    Counters = seshat_counters:new(?MODULE, Labels, ?PROTOCOL_QUEUE_TYPE_COUNTERS),
+    persistent_term:put({?MODULE, Protocol, QueueType}, Counters),
+    ok;
+init(Labels = [{protocol, Protocol}]) ->
+    _ = seshat_counters:new_group(?MODULE),
+    Counters = seshat_counters:new(?MODULE, Labels, ?PROTOCOL_COUNTERS),
+    persistent_term:put({?MODULE, Protocol}, Counters),
+    ok.
+
+overview() ->
+    seshat_counters:overview(?MODULE).
+
+prometheus_format() ->
+    seshat_counters:prometheus_format(?MODULE).
+
+messages_received(Protocol, Num) ->
+    counters:add(fetch(Protocol), ?MESSAGES_RECEIVED, Num).
+
+messages_received_confirm(Protocol, Num) ->
+    counters:add(fetch(Protocol), ?MESSAGES_RECEIVED_CONFIRM, Num).
+
+% Formerly known as queue_messages_published_total
+% This is a more accurate representation of what happens with messages that
+%   arrive on a channel or connection in the case of streams.
```

Comment:
As commented above, in my test, I didn't see that metric getting incremented when sending a message via `stream` protocol.

Reply:
Right, this is not something that a `stream_reader` does. These are all the counters that it currently tracks: https://github.com/rabbitmq/rabbitmq-server/blob/b2200046c474efddba614254760cce10eaab6273/deps/rabbitmq_stream/src/rabbit_stream_reader.erl#L277-L294 . I will remove the comment because it's obviously confusing.